test/lit/passes/type-merging-desc.wast

```diff
@@ -23,53 +23,310 @@
+
+  ;; CHECK:      (func $desc (type $4) (result (ref $B.desc))
+  ;; CHECK-NEXT:  (ref.get_desc $B
+  ;; CHECK-NEXT:   (struct.new_default $B
+  ;; CHECK-NEXT:    (struct.new_default $B.desc)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $desc (result (ref $B.desc))
+    (ref.get_desc $B
+      (struct.new_default $B
+        (struct.new_default $B.desc)
+      )
+    )
+  )
+)
+
+(module
+  ;; We cannot optimize because $B.desc has an extra field.
+  (rec
+    ;; CHECK:      (rec
+    ;; CHECK-NEXT:  (type $A (sub (descriptor $A.desc (struct))))
+    (type $A (sub (descriptor $A.desc (struct))))
+    ;; CHECK:       (type $A.desc (sub (describes $A (struct))))
+    (type $A.desc (sub (describes $A (struct))))
+    ;; CHECK:       (type $B (sub $A (descriptor $B.desc (struct))))
+    (type $B (sub $A (descriptor $B.desc (struct))))
+    ;; CHECK:       (type $B.desc (sub $A.desc (describes $B (struct (field i32)))))
+    (type $B.desc (sub $A.desc (describes $B (struct (field i32)))))
+  )
+
+  ;; CHECK:       (type $4 (func (result (ref $B.desc))))
+
+  ;; CHECK:      (global $A (ref null $A) (ref.null none))
+  (global $A (ref null $A) (ref.null none))
+  ;; CHECK:      (global $A.desc (ref null $A.desc) (ref.null none))
+  (global $A.desc (ref null $A.desc) (ref.null none))
+
+  ;; CHECK:      (func $desc (type $4) (result (ref $B.desc))
+  ;; CHECK-NEXT:  (ref.get_desc $B
+  ;; CHECK-NEXT:   (struct.new_default $B
+  ;; CHECK-NEXT:    (struct.new_default $B.desc)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $desc (result (ref $B.desc))
+    (ref.get_desc $B
+      (struct.new_default $B
+        (struct.new_default $B.desc)
+      )
+    )
+  )
+)
+
+(module
+  ;; We can optimize because $B matches $A and $B.desc matches $A.desc.
+  (rec
+    ;; CHECK:      (rec
+    ;; CHECK-NEXT:  (type $A (sub (descriptor $A.desc (struct))))
+    (type $A (sub (descriptor $A.desc (struct))))
+    ;; CHECK:       (type $A.desc (sub (describes $A (struct))))
+    (type $A.desc (sub (describes $A (struct))))
+    (type $B (sub $A (descriptor $B.desc (struct))))
+    (type $B.desc (sub $A.desc (describes $B (struct))))
+  )
+
+  ;; CHECK:       (type $2 (func (result (ref $A.desc))))
+
+  ;; CHECK:      (global $A (ref null $A) (ref.null none))
+  (global $A (ref null $A) (ref.null none))
+  ;; CHECK:      (global $A.desc (ref null $A.desc) (ref.null none))
+  (global $A.desc (ref null $A.desc) (ref.null none))
+
+  ;; CHECK:      (func $desc (type $2) (result (ref $A.desc))
+  ;; CHECK-NEXT:  (ref.get_desc $A
+  ;; CHECK-NEXT:   (struct.new_default $A
+  ;; CHECK-NEXT:    (struct.new_default $A.desc)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $desc (result (ref $B.desc))
+    (ref.get_desc $B
+      (struct.new_default $B
+        (struct.new_default $B.desc)
+      )
+    )
+  )
+)
+
+(module
+  ;; The two chains have the same sequence of children, but they are divided
```

Comment:
What does "sequence of children" mean here?

Reply:
They have the same child types appearing in the same order.